test/modules/encoders/test_albef_vision_encoder.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from functools import partial
+from typing import OrderedDict
+
+import torch
+from test.test_utils import assert_expected, set_rng_seed
+from torch import nn, Tensor
+from torchmultimodal.modules.encoders.albef_vision_encoder import ALBEFVisionEncoder
+
+
+class TestALBEFVisionEncoder:
+    set_rng_seed(0)
+    input = torch.randn(1, 3, 4, 4)
+    proj_input = torch.randn(1, 1, 3)
+    proj_state_dict = OrderedDict(
+        [("weight", torch.randn(3, 3, 4, 4)), ("bias", torch.randn(3))]
+    )
+    attention_state_dict = OrderedDict(
+        [
+            ("in_proj_weight", torch.randn(9, 3)),
+            ("in_proj_bias", torch.randn(9)),
+            ("out_proj.weight", torch.randn(3, 3)),
+            ("out_proj.bias", torch.randn(3)),
+        ]
+    )
+    mlp_state_dict = OrderedDict(
+        [
+            ("0.weight", torch.randn(6, 3)),
+            ("0.bias", torch.randn(6)),
+            ("3.weight", torch.randn(3, 6)),
+            ("3.bias", torch.randn(3)),
+        ]
+    )
+    encoder_block_state_dict = OrderedDict(
+        [
+            ("ln_1.weight", torch.randn(3)),
+            ("ln_1.bias", torch.randn(3)),
+            ("ln_2.weight", torch.randn(3)),
+            ("ln_2.bias", torch.randn(3)),
+        ]
+        + [("self_attention." + key, val) for key, val in attention_state_dict.items()]
+        + [("mlp." + key, val) for key, val in mlp_state_dict.items()]
+    )
+    vit_state_dict = OrderedDict(
+        [
+            ("class_token", torch.randn(1, 1, 3)),
+            ("encoder.pos_embedding", torch.randn(1, 2, 3)),
+            ("encoder.ln.weight", torch.randn(3)),
+            ("encoder.ln.bias", torch.randn(3)),
+        ]
+        + [("conv_proj." + key, val) for key, val in proj_state_dict.items()]
+        + [
+            ("encoder.layers.encoder_layer_0." + key, val)
+            for key, val in encoder_block_state_dict.items()
+        ]
+        + [
+            ("encoder.layers.encoder_layer_1." + key, val)
+            for key, val in encoder_block_state_dict.items()
+        ]
+    )
+    vision_encoder = ALBEFVisionEncoder(
+        image_size=4,
+        patch_size=4,
+        num_layers=2,
+        num_heads=1,
+        hidden_dim=3,
+        mlp_dim=6,
+        norm_layer=partial(nn.LayerNorm, eps=1e-6),
+        num_classes=3,
+    )
+
+    def test_conv_proj(self):
```

Comment:
I like that you're thoroughly testing each one of the components here. I think some of the cleanup I mention above, plus some comments on the test, will help elucidate this to the code reader